Use bufio.Reader to read log

loganhz opened the pull request:
related issue [rancher/rancher#10009](https://github.com/rancher/rancher/issues/10009)

Some line of kubernetes-agent log is too long. Agent will throw the error `time="2017-10-05T05:33:41Z" level=error msg="Error with the container log scanner." error="bufio.Scanner: token too long`.

Scanning stops unrecoverably at EOF, the first I/O error, or a token too large to fit in the buffer. When a scan stops, the reader may have advanced arbitrarily far past the last token. Programs that need more control over error handling or large tokens, or must run sequential scans on a reader, should use bufio.Reader instead.

vincent99 reviewed (commented):
If there are lines where 64k isn't enough, why is 128k going to be?

loganhz commented:
@vincent99 it's not enough.  I saw a line with 82k in `rancher-kubernetes-agent` log.  It might be helpful at least for it.  or maybe we should use `bufio.Reader`

Commit:
Fix the log issue when the line size is larger than 64k

Scanning stops unrecoverably at EOF, the first I/O error, or a token too large to fit in the buffer. When a scan stops, the reader may have advanced arbitrarily far past the last token. Programs that need more control over error handling or large tokens, or must run sequential scans on a reader, should use bufio.Reader instead.

https://github.com/rancher/rancher/issues/10009

loganhz commented:
@yasker FYI

yasker commented:
@vincent99 PR updated to bufio.Reader.

StrongMonkey commented:
LGTM
